app/src/main/java/de/xikolo/download/DownloadIdentifier.kt

```diff
@@ -0,0 +1,3 @@
+package de.xikolo.download
+
+interface DownloadIdentifier
```

Comment:
That's because anything can be an identifier for a download. I could have as well used `Any` instead of `DownloadIdentifier` throughout the architecture as the generic type. But having a dedicated interface (even if it's empty) makes it easier to extend the architecture in the future while achieving type safety.

Reply:
But then you could define an ID with a generic type here that will be defined by subclasses?

Reply:
Yes, but in the architecture, not the ID is the identifier which is passed around, but the `DownloadIdentifier` object.

Reply:
There could also be multiple parts identifying a download

Reply:
I see, fine.